### PR TITLE
working map prototype

### PR DESCRIPTION
**Pull request by sweeney-th**
This will move the working react app (scaffold-ed with Next.js) into the staging branch.

### CODE REVIEW

**Comment by grantstromgren** [comment]
Hey @sweeney-th - this is cool! Before I start reviewing though just to reduce the footprint, can you blast away the `.next` folder and add it to the `.gitignore` and update this PR? I don't think we need it and it adds an immense amount of files

**Comment by sweeney-th** [comment]
sure!

**Comment by grantstromgren** [comment]
Ok cool, finally had some time to sit and review this:

- [ ] Run `npm install` to create the `package-lock.json` file. This is like the yarn lock file, it gets generated the first time you run `npm install` which is nice to give people options.
- [ ] Add `npm run build` as a step, as we blasted our `.next` folder that is a necessary step to build the app
- [ ] Do we use the `citieslived.json` file? Can we remove it?

File: misc/state-scores.json [hunk 1]
```diff
@@ -0,0 +1 @@
...core": 1}, {"state": "Louisiana", "score": 1}, {"state": "Maine", "score": 4}, {"state": "Maryland", "score": 5}, {"state": "Massachusetts", "score": 2}, {"state": "Michigan", "score": 0}, {"state": "Minnesota", "score": 5}, {"state": "Missouri", "score": 2}, {"state": "Mississippi", "score": 5}, {"state": "Montana", "score": 1}, {"state": "North Carolina", "score": 3}, {"state": "North Dakota", "score": 1}, {"state": "Nebraska", "score": 1}, {"state": "New Hampshire", "score": 4}, {"state": "New Jersey", "score": 0}, {"state": "New Mexico", "score": 3}, {"state": "Nevada", "score": 1}, {"state": "New York", "score": 3}, {"state": "Ohio", "score": 3}, {"state": "Oklahoma", "score": 3}, {"state": "Oregon", "score": 4}, {"state": "Pennsylvania", "score": 3}, {"state": "Rhode Island", "score": 3}, {"state": "South Carolina", "score": 4}, {"state": "South Dakota", "score": 0}, {"state": "Tennessee", "score": 3}, {"state": "Texas", "score": 3}, {"state": "Utah", "score": 5}, {"state": "Virginia", "score": 5}, {"state": "Vermont", "score": 0}, {"state": "Washington", "score": 4}, {"state": "Wisconsin", "score": 2}, {"state": "West Virginia", "score": 1}, {"state": "Wyoming", "score": 2}]
```
Comment (grantstromgren) [inline misc/state-scores.json:1]: NIT: Can we run prettier, beautify, or something on this file to make it a little easier on the eyes?

File: components/UsMap.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+class UsMap extends Component {
+
+  componentDidMount() {
+
+
+    this.statesLived = statesLived;
+    this.stateScores = stateScores;
+    this.usData = usData;
+    this.height = 500; (64, 0, 64)
```
Comment (grantstromgren) [inline components/UsMap.js:16]: Extra `(64, 0, 64)` here?

File: components/UsMap.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+          .duration(200)
+          .style('opacity', 1)
+          .text("Click detected on " + d.properties.name + ", create card here")
+      });
+
+  }
+
+  render() {
+    return <div id={"#" + this.props.id
+    } ></div >
```
Comment (grantstromgren) [inline components/UsMap.js:125]: NIT: No need for a line break here, the whole <div> can be on one line

File: package.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "nextd3",
```
Comment (grantstromgren) [inline package.json:2]: We can add `freefrom-map-frontend` here instead

File: pages/api/hello.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
```
Comment (grantstromgren) [inline pages/api/hello.js:1]: Do we need this file at all? If not let's blast it away

File: pages/index.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import Head from 'next/head'
+import styles from '../styles/Home.module.css'
+import UsMap from '../components/UsMap'
+
+export default function Home() {
+  return (
+    <div>
+      <UsMap></UsMap>
```
Comment (grantstromgren) [inline pages/index.js:8]: NIT: since we only have one component, let's just return that - remove <div>

Comment (grantstromgren) [reply pages/index.js:8]: Unless there is a reason it's nested here of course

File: components/UsMap.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        }
+      }
+    }
+    return usData
+  }
+
+  renderMap() {
+
+    // just unpacking for tidier variable names downstream
+    let usData = this.usData;
```
Comment (grantstromgren) [inline components/UsMap.js:48]: No change needed, but you could write this as destructured like:

`let { usData, statesLived, colorRagen, width, height, stateScores } = this;`

File: components/UsMap.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+      .style("stroke", "#fff")
+      .style("stroke-width", "1")
+      .style("fill", function (d) {
+        return color(d.properties.score)
+      }).on('mouseover', function (d) {
+        d3.select(this)
+          .style({ opacity: '0.75' })
+          .append("text")
+          .text(function (d) {
+            console.log(d.properties.name)
```
Comment (grantstromgren) [inline components/UsMap.js:100]: I think we can remove any `console.log` in the project we don't really need, makes a cleaner prod

**Comment by grantstromgren** [comment]
Can we update the `favicon` with freefrom's favicon? I think the `admin` folder has both the favicon and logo in there, we can also replace the vercel svg

**Review by grantstromgren — CHANGES_REQUESTED** [review]

File: components/UsMap.js (hunk 7 above)
Comment (sweeney-th) [reply components/UsMap.js:48]: Oh, that's nice. I'm new to JS - never seen this, ty.

File: components/UsMap.js (hunk 8 above)
Comment (sweeney-th) [reply components/UsMap.js:100]: Roger!

File: pages/api/hello.js (hunk 5 above)
Comment (sweeney-th) [reply pages/api/hello.js:1]: Hmm - I'm not sure. It was part of the scaffolding from the create command. Comment all the code in it doesn't seem to break anything. I'll look into this.

File: components/UsMap.js (hunk 3 above)
Comment (sweeney-th) [reply components/UsMap.js:125]: Ugly right? That was my linter - it doesn't object when I change it though

File: misc/state-scores.json (hunk 1 above)
Comment (sweeney-th) [reply misc/state-scores.json:1]: I can actually just remove this is you prefer - it's just the output of the script I wrote to generate fake data to dev on which I copied into public/data/ (and got prettified automatically)

File: components/UsMap.js (hunk 2 above)
Comment (sweeney-th) [reply components/UsMap.js:16]: Got it! Weird.

File: package.json (hunk 4 above)
Comment (sweeney-th) [reply package.json:2]: done

**Comment by sweeney-th** [comment]
> Ok cool, finally had some time to sit and review this:
> 
>     * [ ]  Run `npm install` to create the `package-lock.json` file. This is like the yarn lock file, it gets generated the first time you run `npm install` which is nice to give people options.
> 
>     * [ ]  Add `npm run build` as a step, as we blasted our `.next` folder that is a necessary step to build the app
> 
>     * [ ]  Do we use the `citieslived.json` file? Can we remove it?

No, and yes!

File: pages/index.js (hunk 6 above)
Comment (sweeney-th) [reply pages/index.js:8]: Roger - no reason I can think of other than that div was there from the scaffold so I used it

File: pages/api/hello.js (hunk 5 above)
Comment (sweeney-th) [reply pages/api/hello.js:1]: I think this is just an example they include to demonstrate routing. Deleting.

**Comment by sweeney-th** [comment]
> Can we update the `favicon` with freefrom's favicon? I think the `admin` folder has both the favicon and logo in there, we can also replace the vercel svg

Done!

**Comment by sweeney-th** [comment]
I've push the requested changes.

**Review by grantstromgren — APPROVED** [review]
Looks good! Let's get this merged in 👍

**Pull request merged**